### PR TITLE
wsd: correction in conversion  from timestamp to ISO8601

### LINKED ISSUE #168: X-LOOL-WOPI-Timestamp error
Looks like collabora parses milliseconds with error.
For example i sent 2020-09-22T21:45:12.583000Z in CheckFileInfo
but then i recieve 2020-09-22T21:45:12.000583Z in X-LOOL-WOPI-Timestamp in PutFile.

### PR DESCRIPTION
**Pull request by lpranam**
closes #168

Change-Id: Ibdb1aa72a6961d4707c2336382cc593c984d3615

### CODE REVIEW

**Review by Ashod — CHANGES_REQUESTED** [review]
Thank you for the patch. I expect unit-tests that both reproduce the case you hit, and confirm the fix, to be added.

Can you please add a couple of tests to cover the fix?

Thanks!

**Comment by lpranam** [comment]
@Ashod there are already Whitebox tests for this, but the bug was machine-dependent. I don't know how to write a test in this case

**Comment by kendy** [comment]
@lpranam: In this case I' think it's enough to have a test that asserts that 2020-09-22T21:45:12.583000Z (from the #168 ) is parsed correctly now, no need to go deeper; the people on the other platform will see in case it fails (or in case the code is refactored and a bug is brought in).

std::string getIso8601FracformatTime() seem nicely self-contained, so just a simple 2-liner test will be enough :-)

File: test/WhiteBoxTests.cpp [hunk 1]
```diff
@@ -941,6 +941,10 @@ void WhiteBoxTests::testTime()
     t = std::chrono::system_clock::time_point(std::chrono::nanoseconds(1569592993495336798));
     LOK_ASSERT_EQUAL(std::string("Fri, 27 Sep 2019 14:03:13"), Util::getHttpTime(t));
 
+    t = Util::iso8601ToTimestamp("2020-09-22T21:45:12.583000Z", "LastModifiedTime");
+    LOK_ASSERT_EQUAL(std::string("2020-09-22T21:45:12.583000Z"),
```
Comment (Ashod) [inline test/WhiteBoxTests.cpp:945]: Thanks. Can we also have one with "12.583Z" ?

Comment (lpranam) [reply test/WhiteBoxTests.cpp:945]: done @Ashod

**Review by timar — APPROVED** [review]

File: common/Util.cpp [hunk 2]
```diff
@@ -933,8 +933,15 @@ namespace Util
 
         char* end = nullptr;
         const size_t us = strtoul(trailing + 1, &end, 10); // Skip the '.' and read as integer.
+
+        std::size_t denominator = 1;
+        for (std::size_t i = 1; isdigit(*(trailing + i)); i++)
```
Comment (Ashod) [inline common/Util.cpp:938]: We already know the last char* (`end`, from line 934), so only need to iterate from `trailing` to `end`. No need to call `isdigit`.

**Review by Ashod — APPROVED** [review]
Minor note, but looks good otherwise.

File: common/Util.cpp (hunk 2 above)
Comment (lpranam) [reply common/Util.cpp]: corrected.

**Pull request merged**